<?xml version="1.0"?>
<launch>
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find agv)/config/rosconsole.config"/>

<!-- TX2 Server Socket  -->
  <node name="tx2_communication_server" pkg="agv" type="socket_server.py" args="192.168.1.142 8889" output="screen"/>


<!-- Sensor Fusion -->
  <node name="sensor_fusion" pkg="agv" type="sensor_fusion" output="screen">
    
    <!-- Hyperparameters -->
    <param name="sensor_fusion_frequency"     value="20"/>

    <param name="odom_topic"                  value="/odom/filtered"/>
    <param name="obstacle_topic"              value="/obstacles"/>
    <param name="camera_obstacle_topic"       value="/detection_results"/>
    <param name="sensor_fusion_topic"         value="/sensor_fusion_results"/>
    <param name="deprojection_topic"          value="/deprojection_results"/>
  </node>


</launch>